pkg/backend/spinnaker/backend_test.go

```diff
@@ -1176,3 +1176,452 @@ func TestTestingRemoteStringifyFalse(t *testing.T) {
 
 	assert.Nil(t, err)
 }
+
+func TestDeleteSuccess(t *testing.T) {
+	nestedPipelineString := `
+	{
+		"application": "appname",
+		"name":  "Nested pipeline",
+		"stages": [
+			{
+				"application": "appname",
+				"name":  "Nested pipeline stage",
+				"type": "pipeline",
+				"pipeline": {
+					"application": "appname",
+					"name": "Child pipeline 1",
+					"stages": [
+						{
+							"application": "appname",
+							"name":  "Child pipeline stage",
+							"type": "pipeline",
+							"pipeline": {
+								"application": "appname",
+								"name": "Child pipeline 2",
+								"stages": [
+									{
+										"name": "Child pipeline 2 stage",
+										"application": "appname",
+										"type": "pipeline"
+									}
+								]
+							}
+						},
+						{
+							"application": "appname",
+							"name":  "Child pipeline stage 2",
+							"type": "pipeline",
+							"pipeline": {
+								"application": "appname",
+								"name": "Child pipeline 2.2",
+								"type": "pipeline",
+								"stages": [
+									{
+										"name": "Child pipeline 2.2 stage",
+										"application": "appname",
+										"type": "pipeline"
+									}
+								]
+							}
+						}
+					]
+				}
+			}
+		]
+	}
+	`
+
+	// Test
+	res, err := cli.DeletePipeline(nestedPipelineString)
+
+	// Assert
+	assert.Nil(t, err)
+	assert.Equal(t, res.StatusCode, http.StatusOK)
+}
+
+func TestDeleteSimplePipelineSuccess(t *testing.T) {
+	// Given
+	nestedPipelineString := `
+	{
+		"application": "test123",
+		"name": "simple pipeline",
+		"stages": [
+			 {
+					"name": "Wait",
+					"waitTime": 1,
+					"type": "wait"
+			 }
+		]
+ }
+	`
+
+	// Test
+	res, err := cli.DeletePipeline(nestedPipelineString)
+
+	// Assert
+	assert.Nil(t, err)
+	assert.Equal(t, res.StatusCode, http.StatusOK)
+}
+
+func TestMissingApplicationFailedDelete(t *testing.T) {
+	// Given
+	nestedPipelineString := `
+	{
+		"application": "appname",
+		"name":  "Nested pipeline",
+		"stages": [
+			{
+				"application": "appname",
+				"name":  "Nested pipeline stage",
+				"type": "pipeline",
+				"pipeline": {
+					"name": "Child pipeline 1",
+					"stages": [
+						{
+							"name": "Child pipeline 2 stage",
+							"application": "appname",
+							"type": "pipeline"
+						}
+					]
+				}
+			}
+		]
+	}
+	`
+
+	// Test
+	_, err := cli.DeletePipeline(nestedPipelineString)
+
+	// Assert
+	assert.EqualError(t, err, "required pipeline key 'application' missing")
+}
+
+func TestMissingNameFailedDelete(t *testing.T) {
```

Comment:
Stylistic nitpick.

You can combine tests (like all failing ones) into one, with sub-tests inside.

For example, combining both `TestMissingNameFailedDelete` and `TestPipelineChildPipelineWrongApplicationFailedDelete`:

```go
    // Given
    tests := []struct {
        name            string
        pipelineString  string
        errorString     string
    }{
        {
            name: "missing-name",
            pipelineString: "{...}", 
            errorString: "required pipeline key 'name' missing"
        },
        {
            name: "child-ipeline-wrong-application",
            pipelineString: "{...}", 
            errorString: "pipeline 'application' key value should match the value of parent pipeline 'application' key"
        },
    }

    for _, test := range tests {
        t.Run(test.name, func(t *testing.T) {
            // When
            _, err := cli.DeletePipeline(test.pipelineString)

            // Then
            assert.EqualError(t, err, test.errorString)
        })
    }
```

Reply:
Fair enough, there can be another PR/issue to refactor if needed.

This was more of a nitpick/try-and-see suggestion.